### PR TITLE
Reuse class filter between diff and profile tabs.

### PR DESCRIPTION
**Pull request by polina-c**
Fixes: https://github.com/flutter/devtools/issues/7419

https://github.com/flutter/devtools/assets/12115586/4a8a65de-0602-4806-ab97-79457ec39317

### CODE REVIEW

File: packages/devtools_app/test/memory/diff/controller/memory_footprint_test.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import 'dart:io';
+
+import 'package:devtools_app/src/shared/memory/heap_data.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../../test_infra/test_data/memory/heap/heap_data.dart';
+
+void main() {
+  test(
+    'heap snapshot processing creates reasonable memory footprint',
```
Comment (polina-c) [inline packages/devtools_app/test/memory/diff/controller/memory_footprint_test.dart:16]: The file is moved. Test name is the only change in this file.

File: packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart [hunk 2]
```diff
@@ -69,18 +69,32 @@ class MemoryFeatureControllers {
...
       profilePaneController,
     );
+    shareClassFilterBetweenProfileAndDiff();
   }
 
   /// Sub-controllers of memory controller.
   late final MemoryFeatureControllers controllers;
 
+  void shareClassFilterBetweenProfileAndDiff() {
+    controllers.diff.derived.applyFilter(controllers.profile.classFilter.value);
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart:86]: Nit: add a trailing comma to keep this under 80 characters

File: packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart [hunk 3]
```diff
@@ -69,18 +69,32 @@ class MemoryFeatureControllers {
...
+
+    controllers.profile.classFilter.addListener(() {
+      controllers.diff.derived
+          .applyFilter(controllers.profile.classFilter.value);
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart:90]: Ditto

File: packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart [hunk 4]
```diff
@@ -69,18 +69,32 @@ class MemoryFeatureControllers {
...
+    });
+
+    controllers.diff.core.classFilter.addListener(() {
+      controllers.profile.setFilter(controllers.diff.core.classFilter.value);
```
Comment (bkonyi) [inline packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart:94]: Just confirming, setting the filter in these callbacks doesn't trigger the other listener right?

**Review by bkonyi — APPROVED** [review]

File: packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart (hunk 4 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/framework/connected/memory_controller.dart:94]: It does. Profile and diff controllers will do their own handling of the new filter.
But number of classes is not huge, so it is ok.

**Pull request merged**